### PR TITLE
generate kube: do not set caps with --privileged

### PR DESCRIPTION
**Pull request by vrothberg**
Do not play with capabilities for privileged containers where all
capabilities will be set implicitly.

While the initital error report from #8897 is not reproducible,
privilged containers could not be executed due the bug at hand.

Fixes: #8897
Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

**Comment by baude** [comment]
LGTM

**Comment by vrothberg** [comment]
Alright, hitting the same error as reported. Will have another look.

File: libpod/kube.go [hunk 1]
```diff
@@ -625,13 +625,18 @@ func capAddDrop(caps *specs.LinuxCapabilities) (*v1.Capabilities, error) {
...
 func generateKubeSecurityContext(c *Container) (*v1.SecurityContext, error) {
-	priv := c.Privileged()
+	privileged := c.Privileged()
 	ro := c.IsReadOnly()
 	allowPrivEscalation := !c.config.Spec.Process.NoNewPrivileges
 
-	newCaps, err := capAddDrop(c.config.Spec.Process.Capabilities)
-	if err != nil {
-		return nil, err
+	var capabilities *v1.Capabilities
```
Comment (edsantiago) [inline libpod/kube.go:632]: Shouldn't this be initialized? (I'm not a Gohead; I may be missing something obvious)

**Comment by mheon** [comment]
Tests red. Otherwise LGTM

File: libpod/kube.go (hunk 1 above)
Comment (vrothberg) [reply libpod/kube.go:632]: That's no required here. Setting the container to privileged implies that all caps get set later on:
```
$ podman (fix-8897) $ podman top 5e5eb6e0fb43 capeff
EFFECTIVE CAPS
full
```

**Comment by vrothberg** [comment]
This is looking good now :+1:

**Comment by TomSweeneyRedHat** [comment]
LGTM
and all green if you want to cancel the hold @mheon

**Pull request merged**